shell/shared/grain.js

```diff
@@ -590,9 +590,14 @@ if (Meteor.isClient) {
           assignment = call.roleAssignment;
         }
         // Tokens expire by default in 5 minutes from generation date
-        var selfDestructTime = Date.now() + (5 * 60 * 1000);
+        var selfDestructDuration = 5 * 60 * 1000;
+
+        var rawParentToken;
+        if (Router.current().route.getName() === "shared") {
```

Comment:
We dispose the iframe.

Also, we have this check at the top of the function:

    if (event.origin !== currentAppOrigin) {

So I think that should prevent any app's postMessage() from being interpreted as from some other app.